test/e2e/persistent_volumes.go

```diff
@@ -62,34 +77,48 @@ func nfsServerPodCleanup(c clientset.Interface, config VolumeTestConfig) {
 	}
 }
 
-// Cleanup up pvs and pvcs in multi-pv-pvc test cases. All entries found in the pv and
+// Clean up a pv and pvc in a single pv/pvc test case.
+func pvPvcCleanup(c clientset.Interface, ns string, pv *api.PersistentVolume, pvc *api.PersistentVolumeClaim) {
+	deletePersistentVolumeClaim(c, pvc.Name, ns)
+	deletePersistentVolume(c, pv.Name)
+}
+
+// Clean up pvs and pvcs in multi-pv-pvc test cases. All entries found in the pv and
 // claims maps are deleted.
-// Note: this is the only code that deletes PV objects.
-func pvPvcCleanup(c clientset.Interface, ns string, pvols pvmap, claims pvcmap) {
+func pvPvcMapCleanup(c clientset.Interface, ns string, pvols pvmap, claims pvcmap) {
+	for pvcKey := range claims {
+		deletePersistentVolumeClaim(c, pvcKey.Name, ns)
+		delete(claims, pvcKey)
+	}
 
-	if c != nil && len(ns) > 0 {
-		for pvcKey := range claims {
-			_, err := c.Core().PersistentVolumeClaims(pvcKey.Namespace).Get(pvcKey.Name)
-			if !apierrs.IsNotFound(err) {
-				Expect(err).NotTo(HaveOccurred())
-				framework.Logf("   deleting PVC %v ...", pvcKey)
-				err = c.Core().PersistentVolumeClaims(pvcKey.Namespace).Delete(pvcKey.Name, nil)
-				Expect(err).NotTo(HaveOccurred())
-				framework.Logf("   deleted  PVC %v", pvcKey)
-			}
-			delete(claims, pvcKey)
+	for pvKey := range pvols {
+		deletePersistentVolume(c, pvKey)
+		delete(pvols, pvKey)
+	}
+}
+
+// Delete the PV.
+func deletePersistentVolume(c clientset.Interface, pvName string) {
+	if c != nil && len(pvName) > 0 {
+		_, err := c.Core().PersistentVolumes().Get(pvName)
+		if !apierrs.IsNotFound(err) {
+			Expect(err).NotTo(HaveOccurred())
+			framework.Logf("Deleting PersistentVolume %v", pvName)
+			err := c.Core().PersistentVolumes().Delete(pvName, nil)
+			Expect(err).NotTo(HaveOccurred())
 		}
+	}
+}
 
-		for name := range pvols {
-			_, err := c.Core().PersistentVolumes().Get(name)
-			if !apierrs.IsNotFound(err) {
-				Expect(err).NotTo(HaveOccurred())
-				framework.Logf("   deleting PV %v ...", name)
-				err = c.Core().PersistentVolumes().Delete(name, nil)
-				Expect(err).NotTo(HaveOccurred())
-				framework.Logf("   deleted  PV %v", name)
-			}
-			delete(pvols, name)
+// Delete the Claim
+func deletePersistentVolumeClaim(c clientset.Interface, pvcName string, ns string) {
+	if c != nil && len(pvcName) > 0 {
+		_, err := c.Core().PersistentVolumeClaims(ns).Get(pvcName)
```

Comment:
You should delete the claim without checking and then just check for `IsNotFound` error and ignore it - one less API call.